### PR TITLE
feat(eslint-plugin-rules): add eslint-plugin-rules

### PR DESCRIPTION
**Pull request by lfu**
Add eslint-plugin-rules to hold eslint custom rules for Foreman and plugins

### CODE REVIEW

**Comment by lfu** [comment]
Could someone help me to get a tarball for testing?

File: packages/eslint-plugin-rules/README.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+### Create an eslintrc file
+```js
+{
+  "plugins": ["@theforeman/rules"],
+  "rules": {
+    "@theforeman/rules/require-ouiaid": "warn",
+  }
+}
+```
+These components would be checked for ouiaId field if none is specified in the eslintrc file:
```
Comment (jeremylenz) [inline packages/eslint-plugin-rules/README.md:18]:
```suggestion
If not specified in the eslintrc file, these components will be checked for an `ouiaId` prop:
```

File: packages/eslint-plugin-rules/README.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+You can specify what components you want to check against.
+```js
+{
+  "plugins": ["@theforeman/rules"],
+  "rules": {
+    "@theforeman/rules/require-ouiaid": [
+      "warn",
+      "Button",
+      "Table",
```
Comment (jeremylenz) [inline packages/eslint-plugin-rules/README.md:65]: Is the warning level really the first element in the array with all the other components? That seems odd..

Comment (lfu) [reply packages/eslint-plugin-rules/README.md:65]: Yep, that is the [format](https://eslint.org/docs/latest/extend/custom-rules#accessing-options-passed-to-a-rule).

File: packages/eslint-plugin-rules/README.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+### Create an eslintrc file
+```js
+{
+  "plugins": ["@theforeman/rules"],
+  "rules": {
+    "@theforeman/rules/require-ouiaid": "warn",
+  }
+}
+```
+If not specified in the eslintrc file, these components will be checked for an `ouiaId` prop:
```
Comment (MariaAga) [inline packages/eslint-plugin-rules/README.md:18]: Can you add here why we need an `ouiaId` prop in these components, and an example?

Comment (lfu) [reply packages/eslint-plugin-rules/README.md:18]: `ouiaId` field is used for QE's automation tests.

**Review by sjha4 — APPROVED** [review]
I tested this on https://github.com/Katello/katello/pull/10532 and this works great. 
ACK from me post resolution of code comments here👍🏼

File: packages/eslint-plugin-rules/README.md (hunk 3 above)
Comment (lfu) [reply packages/eslint-plugin-rules/README.md:20]: @MariaAga Updated.

Comment (MariaAga) [reply packages/eslint-plugin-rules/README.md:20]: Thanks

**Comment by ekohl** [comment]
Late to the party, but why was this created? How is this different from @theforeman/eslint-plugin-foreman? And if we think this should be included everywhere, why doesn't `eslint-plugin-foreman` depend on this to enforce it everywhere?

For now I've opened https://github.com/theforeman/foreman/pull/9779 to ignore it in packaging so we don't bother users with it.

**Pull request merged**